### PR TITLE
Issue #111

### PR DESCRIPTION
**Pull request by KawashiroNitori**
Compatible with the latest version of PyMongo

### CODE REVIEW

**Comment by twd2** [comment]
Thanks! But why do you add these spaces like

    -              name: str=None, gravatar: str=None):
    +              name: str = None, gravatar: str = None):

O. O ?

**Comment by KawashiroNitori** [comment]
...seems like the reformatting feature of PyCharm

File: vj4/model/document.py [hunk 1]
```diff
@@ -66,38 +65,38 @@ def convert_doc_id(doc_id):
...
-  doc = await coll.find_and_modify(query={'domain_id': domain_id,
-                                          'doc_type': doc_type,
-                                          'doc_id': doc_id},
-                                   update={'$set': kwargs},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id},
+                                       update={'$set': kwargs},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 2]
```diff
@@ -66,38 +65,38 @@ def convert_doc_id(doc_id):
...
+                                'doc_id': doc_id})
 
 
 async def delete_multi(domain_id: str, doc_type: int, **kwargs):
   # TODO(twd2): delete status?
   coll = db.Collection('document')
-  return await coll.remove({'domain_id': domain_id,
-                            'doc_type': doc_type,
-                            **kwargs}, just_one=False)
+  return await coll.delete_one({'domain_id': domain_id,
```
Comment (twd2) [inline vj4/model/document.py]: I think it should be `delete_many`.

File: vj4/model/document.py [hunk 3]
```diff
@@ -118,24 +117,24 @@ def get_multi(*, fields=None, **kwargs):
...
-  doc = await coll.find_and_modify(query={'domain_id': domain_id,
-                                          'doc_type': doc_type,
-                                          'doc_id': doc_id},
-                                   update={'$inc': {key: value}},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id},
+                                       update={'$inc': {key: value}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 4]
```diff
@@ -118,24 +117,24 @@ def get_multi(*, fields=None, **kwargs):
...
-                                          'doc_id': doc_id},
-                                   update={'$inc': {inc_key: inc_value},
-                                           '$set': {set_key: set_value}},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id},
+                                       update={'$inc': {inc_key: inc_value},
+                                               '$set': {set_key: set_value}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 5]
```diff
@@ -144,26 +143,26 @@ def get_multi(*, fields=None, **kwargs):
...
-                                                           '_id': obj_id}}},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id},
+                                       update={'$push': {key: {**kwargs,
+                                                               'content': content,
+                                                               'owner_uid': owner_uid,
+                                                               '_id': obj_id}}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 6]
```diff
@@ -144,26 +143,26 @@ def get_multi(*, fields=None, **kwargs):
...
-  doc = await coll.find_and_modify(query={'domain_id': domain_id,
-                                          'doc_type': doc_type,
-                                          'doc_id': doc_id},
-                                   update={'$pull': {key: {'_id': sub_id}}},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id},
+                                       update={'$pull': {key: {'_id': sub_id}}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 7]
```diff
@@ -188,36 +187,36 @@ def get_multi(*, fields=None, **kwargs):
...
-                                          'doc_id': doc_id,
-                                          key: {'$elemMatch': {'_id': sub_id}}},
-                                   update={'$set': mod},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id,
+                                               key: {'$elemMatch': {'_id': sub_id}}},
+                                       update={'$set': mod},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 8]
```diff
@@ -188,36 +187,36 @@ def get_multi(*, fields=None, **kwargs):
...
-  doc = await coll.find_and_modify(query={'domain_id': domain_id,
-                                          'doc_type': doc_type,
-                                          'doc_id': doc_id},
-                                   update={'$addToSet': {set_key: content}},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id},
+                                       update={'$addToSet': {set_key: content}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 9]
```diff
@@ -188,36 +187,36 @@ def get_multi(*, fields=None, **kwargs):
...
-  doc = await coll.find_and_modify(query={'domain_id': domain_id,
-                                          'doc_type': doc_type,
-                                          'doc_id': doc_id},
-                                   update={'$pull': {set_key: {'$in': contents}}},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id},
+                                       update={'$pull': {set_key: {'$in': contents}}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 10]
```diff
@@ -227,38 +226,38 @@ def get_multi(*, fields=None, **kwargs):
...
-                                   update={'$set': kwargs},
-                                   upsert=True,
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id,
+                                               'uid': uid},
+                                       update={'$set': kwargs},
+                                       upsert=True,
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 11]
```diff
@@ -227,38 +226,38 @@ def get_multi(*, fields=None, **kwargs):
...
-                                    upsert=True,
-                                    new=True)
+  return await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                                'doc_type': doc_type,
+                                                'doc_id': doc_id,
+                                                'uid': uid,
+                                                key: {'$not': {'$eq': if_not}}},
+                                        update={'$set': {key: value, **kwargs}},
+                                        upsert=True,
+                                        return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 12]
```diff
@@ -276,138 +275,138 @@ def get_multi_status(*, fields=None, **kwargs):
...
-                                   upsert=True,
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id,
+                                               'uid': uid,
+                                               key: {'$not': not_expr}},
+                                       update={'$inc': {key: value}},
+                                       upsert=True,
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 13]
```diff
@@ -276,138 +275,138 @@ def get_multi_status(*, fields=None, **kwargs):
...
-                                   update={'$inc': {key: value}},
-                                   upsert=True,
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id,
+                                               'uid': uid},
+                                       update={'$inc': {key: value}},
+                                       upsert=True,
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 14]
```diff
@@ -276,138 +275,138 @@ def get_multi_status(*, fields=None, **kwargs):
...
-                                   upsert=True,
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id,
+                                               'uid': uid},
+                                       update={'$push': {key: value},
+                                               '$inc': {'rev': 1}},
+                                       upsert=True,
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 15]
```diff
@@ -276,138 +275,138 @@ def get_multi_status(*, fields=None, **kwargs):
...
-                                   update={'$inc': {'rev': 1}},
-                                   upsert=True,
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id,
+                                               'uid': uid},
+                                       update={'$inc': {'rev': 1}},
+                                       upsert=True,
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/document.py [hunk 16]
```diff
@@ -276,138 +275,138 @@ def get_multi_status(*, fields=None, **kwargs):
...
-                                           '$inc': {'rev': 1}},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'domain_id': domain_id,
+                                               'doc_type': doc_type,
+                                               'doc_id': doc_id,
+                                               'uid': uid,
+                                               'rev': rev},
+                                       update={'$set': kwargs,
+                                               '$inc': {'rev': 1}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/document.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/domain.py [hunk 17]
```diff
@@ -59,17 +58,17 @@ def get_multi(*, fields=None, **kwargs):
   if 'name' in kwargs:
     validator.check_name(kwargs['name'])
   # TODO(twd2): check kwargs
-  return await coll.find_and_modify(query={'_id': domain_id},
-                                    update={'$set': {**kwargs}},
-                                    new=True)
+  return await coll.find_one_and_update(filter={'_id': domain_id},
+                                        update={'$set': {**kwargs}},
+                                        return_document=True)
```
Comment (twd2) [inline vj4/model/domain.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/domain.py [hunk 18]
```diff
@@ -59,17 +58,17 @@ def get_multi(*, fields=None, **kwargs):
...
 
 async def unset(domain_id, fields):
   # TODO(twd2): check fields
   coll = db.Collection('domain')
-  return await coll.find_and_modify(query={'_id': domain_id},
-                                    update={'$unset': dict((f, '') for f in set(fields))},
-                                    new=True)
+  return await coll.find_one_and_update(filter={'_id': domain_id},
+                                        update={'$unset': dict((f, '') for f in set(fields))},
+                                        return_document=True)
```
Comment (twd2) [inline vj4/model/domain.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/domain.py [hunk 19]
```diff
@@ -79,9 +78,9 @@ def get_multi(*, fields=None, **kwargs):
     if domain['_id'] == domain_id:
       raise error.BuiltinDomainError(domain_id)
   coll = db.Collection('domain')
-  return await coll.find_and_modify(query={'_id': domain_id},
-                                    update={'$set': {'roles.{0}'.format(role): perm}},
-                                    new=True)
+  return await coll.find_one_and_update(filter={'_id': domain_id},
+                                        update={'$set': {'roles.{0}'.format(role): perm}},
+                                        return_document=True)
```
Comment (twd2) [inline vj4/model/domain.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/domain.py [hunk 20]
```diff
@@ -100,10 +99,10 @@ def get_multi(*, fields=None, **kwargs):
...
                          {'$unset': {'role': ''}}, multi=True)
   coll = db.Collection('domain')
-  return await coll.find_and_modify(query={'_id': domain_id},
-                                    update={'$unset': dict(('roles.{0}'.format(role), '')
-                                                           for role in roles)},
-                                    new=True)
+  return await coll.find_one_and_update(filter={'_id': domain_id},
+                                        update={'$unset': dict(('roles.{0}'.format(role), '')
+                                                               for role in roles)},
+                                        return_document=True)
```
Comment (twd2) [inline vj4/model/domain.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/domain.py [hunk 21]
```diff
@@ -112,9 +111,9 @@ def get_multi(*, fields=None, **kwargs):
     if domain['_id'] == domain_id:
       raise error.BuiltinDomainError(domain_id)
   coll = db.Collection('domain')
-  return await coll.find_and_modify(query={'_id': domain_id, 'owner_uid': old_owner_uid},
-                                    update={'$set': {'owner_uid': new_owner_uid}},
-                                    new=True)
+  return await coll.find_one_and_update(filter={'_id': domain_id, 'owner_uid': old_owner_uid},
+                                        update={'$set': {'owner_uid': new_owner_uid}},
+                                        return_document=True)
```
Comment (twd2) [inline vj4/model/domain.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/domain.py [hunk 22]
```diff
@@ -125,18 +124,18 @@ def get_multi(*, fields=None, **kwargs):
...
 async def set_user(domain_id, uid, **kwargs):
   coll = db.Collection('domain.user')
-  return await coll.find_and_modify(query={'domain_id': domain_id, 'uid': uid},
-                                    update={'$set': kwargs},
-                                    upsert=True,
-                                    new=True)
+  return await coll.find_one_and_update(filter={'domain_id': domain_id, 'uid': uid},
+                                        update={'$set': kwargs},
+                                        upsert=True,
+                                        return_document=True)
```
Comment (twd2) [inline vj4/model/domain.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/domain.py [hunk 23]
```diff
@@ -125,18 +124,18 @@ def get_multi(*, fields=None, **kwargs):
...
 async def unset_user(domain_id, uid, fields):
   coll = db.Collection('domain.user')
-  return await coll.find_and_modify(query={'domain_id': domain_id, 'uid': uid},
-                                    update={'$unset': dict((f, '') for f in set(fields))},
-                                    upsert=True,
-                                    new=True)
+  return await coll.find_one_and_update(filter={'domain_id': domain_id, 'uid': uid},
+                                        update={'$unset': dict((f, '') for f in set(fields))},
+                                        upsert=True,
+                                        return_document=True)
```
Comment (twd2) [inline vj4/model/domain.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/domain.py [hunk 24]
```diff
@@ -177,10 +176,10 @@ def get_multi(*, fields=None, **kwargs):
...
 async def inc_user(domain_id, uid, **kwargs):
   coll = db.Collection('domain.user')
-  return await coll.find_and_modify(query={'domain_id': domain_id, 'uid': uid},
-                                    update={'$inc': kwargs},
-                                    upsert=True,
-                                    new=True)
+  return await coll.find_one_and_update(filter={'domain_id': domain_id, 'uid': uid},
+                                        update={'$inc': kwargs},
+                                        upsert=True,
+                                        return_document=True)
```
Comment (twd2) [inline vj4/model/domain.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/fs.py [hunk 25]
```diff
@@ -77,9 +77,9 @@
 async def unlink(file_id: objectid.ObjectId):
   """Unlink a file."""
   coll = db.Collection('fs.files')
-  doc = await coll.find_and_modify(query={'_id': file_id},
-                                   update={'$inc': {'metadata.link': -1}},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'_id': file_id},
+                                       update={'$inc': {'metadata.link': -1}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/fs.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/message.py [hunk 26]
```diff
@@ -41,28 +41,28 @@ def get_multi(uid: int, *, fields=None):
            'content': content,
            'status': 0,
            'at': datetime.datetime.utcnow()}
-  mdoc = await coll.find_and_modify(query={'_id': message_id},
-                                    update={'$push': {'reply': reply}},
-                                    new=True)
+  mdoc = await coll.find_one_and_update(filter={'_id': message_id},
+                                        update={'$push': {'reply': reply}},
+                                        return_document=True)
```
Comment (twd2) [inline vj4/model/message.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/opcount.py [hunk 27]
```diff
@@ -7,46 +7,45 @@
...
-                                     update={'$inc': {op: operations}},
-                                     upsert=True,
-                                     new=True)
+    doc = await coll.find_one_and_update(filter={'ident': ident,
+                                                 'begin_at': begin_at,
+                                                 'expire_at': expire_at,
+                                                 op: {'$not': {'$gte': max_operations}}},
+                                         update={'$inc': {op: operations}},
+                                         upsert=True,
+                                         return_document=True)
```
Comment (twd2) [inline vj4/model/opcount.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/opcount.py [hunk 28]
```diff
@@ -58,12 +57,12 @@
...
-                                          'expire_at': expire_at},
-                                   update={'$inc': {op: operations}},
-                                   upsert=True,
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'ident': ident,
+                                               'begin_at': begin_at,
+                                               'expire_at': expire_at},
+                                       update={'$inc': {op: operations}},
+                                       upsert=True,
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/opcount.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/record.py [hunk 29]
```diff
@@ -53,19 +52,19 @@
...
+                                                          'judge_at': '',
+                                                          'compiler_texts': '',
+                                                          'judge_texts': '',
+                                                          'cases': ''},
+                                               '$set': {'status': constant.record.STATUS_WAITING,
+                                                        'score': 0,
+                                                        'time_ms': 0,
+                                                        'memory_kb': 0,
+                                                        'rejudged': True}},
+                                       return_document=False)
```
Comment (twd2) [inline vj4/model/record.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/record.py [hunk 30]
```diff
@@ -131,65 +130,65 @@ def get_user_in_problem_multi(uid: int, domain_id: str, pid: document.convert_do
...
+  doc = await coll.find_one_and_update(filter={'_id': record_id},
+                                       update={'$set': {'status': status,
+                                                        'judge_uid': judge_uid,
+                                                        'judge_token': judge_token,
+                                                        'judge_at': datetime.datetime.utcnow(),
+                                                        'compiler_texts': [],
+                                                        'judge_texts': [],
+                                                        'cases': [],
+                                                        'progress': 0.0}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/record.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/record.py [hunk 31]
```diff
@@ -131,65 +130,65 @@ def get_user_in_problem_multi(uid: int, domain_id: str, pid: document.convert_do
...
-  doc = await coll.find_and_modify(query={'_id': record_id,
-                                          'judge_uid': judge_uid,
-                                          'judge_token': judge_token},
-                                   update=kwargs,
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'_id': record_id,
+                                               'judge_uid': judge_uid,
+                                               'judge_token': judge_token},
+                                       update=kwargs,
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/record.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/record.py [hunk 32]
```diff
@@ -131,65 +130,65 @@ def get_user_in_problem_multi(uid: int, domain_id: str, pid: document.convert_do
...
+  doc = await coll.find_one_and_update(filter={'_id': record_id,
+                                               'judge_uid': judge_uid,
+                                               'judge_token': judge_token},
+                                       update={'$set': {'status': status,
+                                                        'score': score,
+                                                        'time_ms': time_ms,
+                                                        'memory_kb': memory_kb},
+                                               '$unset': {'judge_token': '',
+                                                          'progress': ''}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/record.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/system.py [hunk 33]
```diff
@@ -10,19 +10,19 @@
...
   """
   coll = db.Collection('system')
-  doc = await coll.find_and_modify(query={'_id': 'user_counter'},
-                                   update={'$inc': {'value': 1}},
-                                   upsert=True,
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'_id': 'user_counter'},
+                                       update={'$inc': {'value': 1}},
+                                       upsert=True,
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/system.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/token.py [hunk 34]
```diff
@@ -95,12 +95,12 @@ def _get_id(id_binary):
...
   now = datetime.datetime.utcnow()
-  doc = await coll.find_and_modify(
-    query={'_id': _get_id(id_binary), 'token_type': token_type},
+  doc = await coll.find_one_and_update(
+    filter={'_id': _get_id(id_binary), 'token_type': token_type},
     update={'$set': {**kwargs,
                      'update_at': now,
                      'expire_at': now + datetime.timedelta(seconds=expire_seconds)}},
-    new=True)
+    return_document=True)
```
Comment (twd2) [inline vj4/model/token.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/user.py [hunk 35]
```diff
@@ -125,10 +124,10 @@ def get_multi(*, fields=PROJECTION_VIEW, **kwargs):
...
   salt = pwhash.gen_salt()
   coll = db.Collection('user')
-  doc = await coll.find_and_modify(query={'_id': uid},
-                                   update={'$set': {'salt': salt,
-                                                    'hash': pwhash.hash_vj4(password, salt)}},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'_id': uid},
+                                       update={'$set': {'salt': salt,
+                                                        'hash': pwhash.hash_vj4(password, salt)}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/user.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/user.py [hunk 36]
```diff
@@ -148,18 +147,18 @@ def get_multi(*, fields=PROJECTION_VIEW, **kwargs):
...
-                                          'hash': doc['hash']},
-                                   update={'$set': {'salt': salt,
-                                                    'hash': pwhash.hash_vj4(password, salt)}},
-                                   new=True)
+  doc = await coll.find_one_and_update(filter={'_id': doc['_id'],
+                                               'salt': doc['salt'],
+                                               'hash': doc['hash']},
+                                       update={'$set': {'salt': salt,
+                                                        'hash': pwhash.hash_vj4(password, salt)}},
+                                       return_document=True)
```
Comment (twd2) [inline vj4/model/user.py]: return_document=ReturnDocument.AFTER ?

File: vj4/model/user.py [hunk 37]
```diff
@@ -148,18 +147,18 @@ def get_multi(*, fields=PROJECTION_VIEW, **kwargs):
...
   return doc
 
 
 async def set_by_uid(uid, **kwargs):
   coll = db.Collection('user')
-  doc = await coll.find_and_modify(query={'_id': uid}, update={'$set': kwargs}, new=True)
+  doc = await coll.find_one_and_update(filter={'_id': uid}, update={'$set': kwargs}, return_document=True)
```
Comment (twd2) [inline vj4/model/user.py]: return_document=ReturnDocument.AFTER ?

**Pull request merged**